codex/sales/salesagent.nim

```diff
@@ -132,7 +132,7 @@ proc subscribe*(agent: SalesAgent) {.async.} =
   await agent.subscribeCancellation()
   agent.subscribed = true
 
-proc unsubscribe*(agent: SalesAgent) {.async.} =
+proc unsubscribe*(agent: SalesAgent) {.async: (raises: [CancelledError]).} =
```

Comment:
This routine as well should not be cancellable imo, since it invokes teardown mechanisms.
```suggestion
proc unsubscribe*(agent: SalesAgent) {.async: (raises: []).} =
```